source/fuzz/fuzzer_pass_inline_functions.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2020 André Perez Maselco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/fuzzer_pass_inline_functions.h"
+
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+#include "source/fuzz/transformation_inline_function.h"
+
+namespace spvtools {
+namespace fuzz {
+
+FuzzerPassInlineFunctions::FuzzerPassInlineFunctions(
+    opt::IRContext* ir_context, TransformationContext* transformation_context,
+    FuzzerContext* fuzzer_context,
+    protobufs::TransformationSequence* transformations)
+    : FuzzerPass(ir_context, transformation_context, fuzzer_context,
+                 transformations) {}
+
+FuzzerPassInlineFunctions::~FuzzerPassInlineFunctions() = default;
+
+void FuzzerPassInlineFunctions::Apply() {
+  std::vector<opt::Instruction*> function_call_instructions;
+
+  for (auto& function : *GetIRContext()->module()) {
+    for (auto& block : function) {
+      for (auto& instruction : block) {
+        if (!GetFuzzerContext()->ChoosePercentage(
+                GetFuzzerContext()->GetChanceOfInliningFunction())) {
+          continue;
+        }
+
+        // |instruction| must be OpFunctionCall to consider applying the
+        // transformation.
+        if (instruction.opcode() != SpvOpFunctionCall) {
+          continue;
+        }
+
+        // |called_function| must not have an early return.
+        auto called_function = fuzzerutil::FindFunction(
+            GetIRContext(), instruction.GetSingleWordInOperand(0));
+        if (called_function->HasEarlyReturn()) {
+          continue;
+        }
+
+        // Checks if some block in |called_function| has OpKill or OpUnreachable
+        // as its termination instruction.
+        bool hasKillOrUnreachableInstruction = false;
+        for (auto& called_function_block : *called_function) {
```

Comment:
See [Google C++ style](https://github.com/KhronosGroup/SPIRV-Tools/pull/3728) - variable names should be lower case with underscores between words.